src/plugins/camera/camera_impl.cpp

```diff
@@ -1583,4 +1612,117 @@ void CameraImpl::format_storage_async(Camera::ResultCallback callback)
         cmd_format, std::bind(&CameraImpl::receive_command_result, this, _1, callback));
 }
 
+std::pair<Camera::Result, std::vector<Camera::CaptureInfo>>
+CameraImpl::list_photos(Camera::PhotosRange photos_range)
+{
+    auto prom = std::make_shared<
+        std::promise<std::pair<Camera::Result, std::vector<Camera::CaptureInfo>>>>();
```

Comment:
`prom` doesn't need to be a `shared_ptr` here, right? Because we don't go out of scope until the prom has been set.